### PR TITLE
PT-449 fix invitations error

### PR DESCRIPTION
**Pull request by franck-fourel**
This error occurs when at least 2 invitations are sent to an account that does not yet exist. 
The first invitation can be approved but the second one is stuck

### CODE REVIEW

File: api/invite.go [hunk 1]
```diff
@@ -203,9 +203,19 @@ func (a *Api) AcceptInvite(res http.ResponseWriter, req *http.Request, vars map[
 
 		validationErrors := []error{}
 
+		inviteeDetails, err := a.sl.GetUser(inviteeID, a.sl.TokenProvide())
```
Comment (BigFab) [inline api/invite.go:210]: Est-ce-qu'il ne serait pas temps de mettre 1 peu des commentaires dans notre code ? Pour expliquer ce qui se passe en bon anglais ? Perso, ça m'aide beaucoup de lire en littéraire ce qui se passe dans le code que j'ai sous les yeux :)

**Review by BigFab — CHANGES_REQUESTED** [review]
Apart from the comments in the code that I would like to see, I am OK with that fix

File: api/invite.go (hunk 1 above)
Comment (franck-fourel) [reply api/invite.go:210]: added some comments in bce9e3f

File: models/confirmation.go [hunk 2]
```diff
@@ -146,7 +146,18 @@ func (c *Confirmation) ValidateCreatorID(expectedCreatorID string, validationErr
...
+// ValidateUserID
+// Validate the user based on UserID and Email (entered as username)
+//
+// expectedUserID: userid of the account
+// expectedEmail: username of the account (email)
+// validationErrors: array of error filled in case of error, empty if everything is ok
+//
+func (c *Confirmation) ValidateUserID(expectedUserID string, expectedEmail string, validationErrors *[]error) *Confirmation {
+
+	if expectedEmail == c.Email && expectedUserID != c.UserId {
```
Comment (BigFab) [inline models/confirmation.go:158]:
```suggestion
	// if a confirmation was already created for the same email in the past, it is reused for the new request
	if expectedEmail == c.Email && expectedUserID != c.UserId {
```

Comment (franck-fourel) [reply models/confirmation.go:158]: it's not exactly the case. The first invitation is accepted with all the required information, but the next ones don't have any UserID, that's the reason why we are forcing to use the one that comes with the email.

**Review by BigFab — APPROVED** [review]
very good

**Pull request merged**